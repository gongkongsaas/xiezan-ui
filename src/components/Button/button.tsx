import { defineComponent, h, PropType } from 'vue'

export type ButtonSize = 'lg' | 'sm'

function lgBtn() {
  return <button>大按钮</button>
}

function smBtn() {
  return <button>小按钮</button>
}

export default defineComponent({
  props: {
    size: {
      type: String as PropType<ButtonSize>,
      required: true,
    },
  },
  setup(props) {
    return () => {
      if (props.size === 'lg') {
        return lgBtn()
      } else {
        return smBtn()
      }
    }
  },
})
