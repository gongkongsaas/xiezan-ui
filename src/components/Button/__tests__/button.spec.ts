import { shallowMount } from '@vue/test-utils'
import Button from '../button'

describe('Button', () => {
  it('lg Button mounted', () => {
    const wrapper = shallowMount(Button, {
      props: {
        size: 'lg',
      },
    } as any)

    expect(wrapper.html()).toMatchSnapshot()
  })
  it('sm Button mounted', () => {
    const wrapper = shallowMount(Button, {
      props: {
        size: 'sm',
      },
    } as any)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
