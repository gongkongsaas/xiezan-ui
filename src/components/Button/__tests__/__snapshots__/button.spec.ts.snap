// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button lg Button mounted 1`] = `<button>大按钮</button>`;

exports[`Button sm Button mounted 1`] = `<button>小按钮</button>`;
